solutions/usaco-860.mdx

```diff
@@ -13,6 +13,131 @@ author: Nathan Gong
 
 <LanguageSection>
 
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int n, k;
+const int MAX_ROWS = 100;
+const int MAX_COLS = 10;
+char board[MAX_ROWS][MAX_COLS];
+vector<vector<bool>> visited(MAX_ROWS, vector<bool>(MAX_COLS, false));
+
+// stores current component indices to set to '0'
+vector<pair<int, int>> component;
+
+
+/*
+ * floodfill to count connected regions
+ * returns size of connected region
+ * fills 'component' vector with the indices of the component
+ */
+int floodfill(int i, int j, char color) {
+	int cnt = 1;
+	if (
+		i < 0 || j < 0 || i >= n || j >= MAX_COLS ||
+		visited[i][j] ||
+		board[i][j] != color
+	) {
+		return 0;
+	}
+	visited[i][j] = 1;
+	component.push_back({i, j});
+	cnt += (
+		floodfill(i + 1, j, color)
+		+ floodfill(i - 1, j, color)
+		+ floodfill(i, j + 1, color)
+		+ floodfill(i, j - 1, color)
+	);
+	
+	return cnt;
+}
+
+int main() {
+	freopen("mooyomooyo.in", "r", stdin);
+	freopen("mooyomooyo.out", "w", stdout);
+
+	cin >> n >> k;
+
+	for (int i = n - 1; i >= 0; i--) {
+		string nums;
+		cin >> nums;
+		for (int j = 0; j < MAX_COLS; j++) {
+			board[i][j] = nums[j];
+		}
+	}
+
+	/*
+	 * Loop while there are regions that got destroyed (board changed)
+	 * If no regions were destroyed, we are at the final state (break loop; return answer)
+	 */
+	bool boardChanged = true;
+	while (boardChanged) {
```

Comment:
(optional) most of the c++ here uses snake_case, so maybe use that in your own code as well?